frontend/src/components/applyForm/README.md

```diff
@@ -0,0 +1,78 @@
+# Apply form component
+
+This component takes form and UI schemas to create a form. The form schema conforms to JSON Schema and can take any shape as long as it conforms to JSON Schema.
+
+[AjV](https://ajv.js.org) is used to validate the schema when saved.
+
+The UI schema is custom for this project. It was inspired by [react-jsonschema-form](https://github.com/rjsf-team/react-jsonschema-form). RJSF was not chosen because it doesn't work with server actions, it doesn't support hierarchical UI layouts natively, and the validation will move to the API which reduces the value of the tightly coupled validation in RJSF. There is a [plugin for layouts](https://github.com/audibene-labs/react-jsonschema-form-layout), however it hasn't been updated in 4 years and still requires a lot of abstraction. This component provides most of the UI rendering with a ~100 line recursive function and uses the same library for validation. The widgets however are patterned after the RJSF widgets.
+
+## UI Schema
+
+The UI Schema is an array of fields or sections.
+
+### Fields
+
+Fields can either reference a field in the form schema:
+
+```
+{
+    type: "field",
+    definition: "/properties/TestField",
+}
+```
+
+or provide its own definitions that can supplement or override the form schema:
+
+```
+{
+    type: "field",
+    definition: "/properties/TestField",
+    schema: {
+        description: "This one is different"
+    }
+}
+```
+
+This can be used for descriptions or enums, but the form will fail if the UI Schema tries to change the field type as the validation is still determined by the form schema.
+
+A field can only exist with the `schema` key in the UI schema, however it won't be saved:
```

Comment:
I'm not sure what this means - is the idea that you can provide a field in the schema that is informational or internal and that won't be saved when the form is saved or submitted?

Reply:
Thanks for asking. I updated to clarify:

> A field can have a schema without a definition. However, this field will only exist in the UI. When the form is saved the field will not be included in the form data sent to the server.

Does that help?